### ISSUE TITLE
A DateTimePicker control should support state announcing on Calendar opening

### ISSUE THREAD

**Issue by DmitryGorokhov** [dotnet/winforms#5487]
* More info:
Not regression. This is an addition functionality for improving experience.
This opportunity appears after adding support Expand-Collapse pattern for DateTimePicker control.

* Linked items:
       - #4099 Issue
       - #5245 PR 

* .NET Core Version: 
6.0.100-rc.2.21419.27

* Have you experienced this same bug with .NET Framework?: 
Yes 

**Problem description:**

An DateTimePicker control should announce state on Calendar opening.

**Expected behavior:**

An DateTimePicker control announce state on Calendar opening action. You can take a look at ComboBox control as an example.

**Minimal repro:**

1. Create a winforms project with a DateTimePicker control.
2. Build and run the project.
3. Launch Narrator (and Narrator Buddy optional).
4. Expand calendar using arrow-button. Collapse calendar using arrow-button or select any value in calendar.

You can detect that date-time-picker don't provide state on open calendar and provide on close. So you task to fix it.

![image](https://user-images.githubusercontent.com/58004471/130207452-463aa09c-0c20-4d53-a040-4a98c3222b53.png)

**Comment by vladimir-krestov** [comment]
@Olina-Zhang, could you please test this issue? It seems, that #6620 fixed this issue and we can close it.
Please use F4 button or mouse to expand the calendar.
As I understand, it's expected behavior:

![image](https://user-images.githubusercontent.com/49272759/159918145-b1f6209d-45ac-432e-9234-efe4e3bcb9ef.png)

Please compare with previous versions.

**Comment by Olina-Zhang** [comment]
Verified this issue in the latest .Net 7.0 build, it is fixed: when expand datetimepicker calendar by F4, Alt+Down or mouse, Narrator can announce the expand state now.
![image](https://user-images.githubusercontent.com/26474449/160337284-c101a43e-fc31-40b2-a1f8-f718e1e1b707.png)
